www/PSPDFKit.js

```diff
@@ -1,244 +1,282 @@
+  channels[key].onHasSubscribersChange = onEventSubscribersChanged;
+}
 
-  for (var key in channels) {
-    console.log("subscriber listener for " + key);
-    channels[key].onHasSubscribersChange = onEventSubscribersChanged;
+/**
+ * Retrieves a named property from the given target object while removing the property from the object.
+ */
+function getPropertyAndUnset(target, name) {
+  var value = target.hasOwnProperty(name) ? target[name] : null;
+  delete target[name];
+  return value;
+}
+
+exports.addEventListener = function(eventType, f) {
+  if (eventType in channels) {
+    channels[eventType].subscribe(f);
   }
+};
 
-  /**
-   * Retrieves a named property from the given target object while removing the property from the object.
-   */
-  function getPropertyAndUnset(target, name) {
-    var value = target.hasOwnProperty(name) ? target[name] : null;
-    delete target[name];
-    return value;
+exports.removeEventListener = function(eventType, f) {
+  if (eventType in channels) {
+    channels[eventType].unsubscribe(f);
   }
+};
 
-  exports.addEventListener = function(eventType, f) {
-    if (eventType in channels) {
-      channels[eventType].subscribe(f);
-    }
-  };
+/**
+ * Adds a new annotation to the current document.
+ *
+ * @param annotation InstantJson of the annotation to add.
+ * @param success   Success callback function.
+ * @param error     Error callback function.
+ */
+exports.addAnnotation = function(annotation, success, error) {
+  exec(success, error, "PSPDFKitCordovaPlugin", "addAnnotation", [annotation]);
+};
 
-  exports.removeEventListener = function(eventType, f) {
-    if (eventType in channels) {
-      channels[eventType].unsubscribe(f);
-    }
-  };
+/**
+ * Applies the passed in document instant json.
+ *
+ * @param annotations The document instant json to apply.
+ * @param success   Success callback function.
+ * @param error     Error callback function.
+ */
+exports.applyInstantJson = function(annotations, success, error) {
+  exec(success, error, "PSPDFKitCordovaPlugin", "applyInstantJson", [
+    annotations
+  ]);
+};
 
+/**
+ * Gets all annotations of the given type from the page.
+ *
+ * @param pageIndex The page to get the annotations for.
+ * @param type The type of annotations to get (See here for types https://pspdfkit.com/guides/server/current/api/json-format/) or null to get all annotations.
+ * @param success   Success callback function.
+ * @param error     Error callback function.
+ */
+exports.getAnnotations = function(pageIndex, type, success, error) {
+  exec(success, error, "PSPDFKitCordovaPlugin", "getAnnotations", [
+    pageIndex,
+    type
+  ]);
+};
+
+/**
+ * Gets all unsaved changes to annotations.
+ *
+ * @param success   Success callback function.
+ * @param error     Error callback function.
+ */
+exports.getAllUnsavedAnnotations = function(success, error) {
+  exec(success, error, "PSPDFKitCordovaPlugin", "getAllUnsavedAnnotations", []);
+};
```

Comment:
And of code block with newly added methods. The rest of the file is reformatting.